ddtrace/span.py

```diff
@@ -258,6 +258,10 @@ def set_tag(self, key, value=None):
         except Exception:
             log.warning("error setting tag %s, ignoring it", key, exc_info=True)
 
+    def set_str_tag(self, key, value):
```

Comment:
Not sure I love this being apart of the public API.

Should we instead name as `_set_str_tag`?

I think in most cases we'll want customers to use `set_tag`, since we have a handful of special tag keys that they use to control functionality.

Reply:
I wonder if we'd benefit from a dict-tag setting helper:

```
def _set_str_tags(self, tags):
    self.meta.update({key: stringify(value) for key, value in tags.items()})
``

Wonder if that is actually any faster than individual `self.meta[name] = stringify(value)` (probably not, since `stringify` is likely the largest overhead there?)

Reply:
Yep, good point. Let's leave it out of the public API until there's a proven demand for it.

So far from what I've seen we don't have a need for `_set_str_tags` internally yet. I'd hold off until we actually need it.

Reply:
> we don't have a need for _set_str_tags internally yet

Agreed, my suggestion was more of a pre-optimization musing more than anything else.